### PR TITLE
DESK-983 DESK-982 DESK-544  Separate out enabled and connected states from active state so that c…

### PR DESCRIPTION
**Pull request by JamieRuderman**
### Description

 Separate out enabled and connected states from active state so that cloud close events don't disconnect

### Test plan

1. Sign in
2. Go to a service and set to proxy only
3. Click connect and wait 15 minutes for disconnect event to come in
4. You should stay connected

### Ticket

<https://remoteit.atlassian.net/browse/DESK-983>
<https://remoteit.atlassian.net/browse/DESK-982>
<https://remoteit.atlassian.net/browse/DESK-544>

### CODE REVIEW

File: frontend/src/buttons/DisconnectButton/DisconnectButton.tsx [hunk 1]
```diff
@@ -13,16 +13,17 @@ type Props = {
...
-  const hidden = !connection || !connection.active
-  const connecting = !!connection?.connecting
+  const hidden = !connection || !connection.connected
+  const disconnecting = !connection?.enabled && !!connection?.connected
   return (
     <Fade in={!hidden} timeout={600}>
       <div>
         <DynamicButton
-          title={connecting ? 'Connecting' : 'Disconnect'}
+          title={disconnecting ? 'Stopping' : 'Disconnect'}
```
Comment (mauricio0000) [inline frontend/src/buttons/DisconnectButton/DisconnectButton.tsx:22]: are you sure to hide the message `Connecting`?

File: src/TrayMenu.ts [hunk 2]
```diff
@@ -84,10 +84,10 @@ export default class TrayMenu {
 
   private connectionsMenu() {
     let menu = []
-    const active = this.pool.filter(c => c.active)
-    const recent = this.pool.filter(c => !c.active)
-    if (active.length) menu.push({ label: 'Active connections', enabled: false }, ...this.connectionsList(active))
-    if (active.length && recent.length) menu.push({ type: 'separator' })
+    const connected = this.pool.filter(c => c.connected)
+    const recent = this.pool.filter(c => !c.connected)
```
Comment (mauricio0000) [inline src/TrayMenu.ts:88]: how you know that every device not connected is recent?

File: frontend/src/buttons/DisconnectButton/DisconnectButton.tsx (hunk 1 above)
Comment (JamieRuderman) [reply frontend/src/buttons/DisconnectButton/DisconnectButton.tsx:22]: Yes - it appears in the other button.

File: src/TrayMenu.ts (hunk 2 above)
Comment (JamieRuderman) [reply src/TrayMenu.ts:88]: That's just where we're putting the disconnected connections. A connection object is only created on connection. This logic isn't changing just the variable names.

**Review by codemodify — APPROVED** [review]

**Pull request merged**